api/src/helperClasses.js

```diff
@@ -20,6 +20,18 @@ class BigQueryParser {
   constructor(mapping){
     this.mapping = mapping;
   }
+// https://stackoverflow.com/questions/2218999/how-to-remove-all-duplicates-from-an-array-of-objects
+// this will only ever iterate over a max of 1000 rows, so it won't tank endpoint speed
+// like totally reformatting the query to deduplicate all the data will
+  deduplicateData (rows) {
+    const dedupe = rows.filter((row, index) => {
+      const _row = JSON.stringify(row);
+      return index === rows.findIndex(obj => {
```

Comment:
This is an odd implementation, deduping should be an O(N) operation and should use a hash table somewhere. Either a set or an object or a map and use the keys to deduplicate.

That being said, I thought we saw a clear fix? Why do you think changing the query would be so much slower? Generally speaking, it's always faster to do something at database level. 

Also, as a consumer, getting the first 1000 slightly faster but getting the whole bucket slower doesn't help! I want the whole thing as fast as possible.